src/coin_test/data/dataset_generator.py

```diff
@@ -27,24 +29,99 @@ def generate(
         """
 
 
-class ReturnsDatasetGenerator(DatasetGenerator):
-    """Create synthetic datasets by shuffling the percentage gains each day."""
+class StitchedChunkDatasetGenerator(DatasetGenerator):
+    """Synthetic Dataset Generator with chunks of data."""
 
     DATASET_TYPE = CustomDataset
 
-    def __init__(self, dataset: "ReturnsDatasetGenerator.DATASET_TYPE") -> None:
-        """Initialize a ResultsDatasetGenerator object."""
+    def __init__(
+        self,
+        dataset: "StitchedChunkDatasetGenerator.DATASET_TYPE",
+        chunk_size: int = 10,
+    ) -> None:
+        """Initialize a chunk synthetic dataset generator."""
         self.dataset = dataset
         self.start: pd.Period = dataset.df.index[0]  # type: ignore
         self.metadata = dataset.metadata
+        self.chunk_size = chunk_size
+
+        if chunk_size < 1:
+            raise ValueError("Chunk size must be a positive integer")
+        elif len(self.dataset.df) < chunk_size:
+            raise ValueError("Chunk size mustn't be larger than the dataset")
+
+    @staticmethod
+    def create_index(
+        start: pd.Period, timedelta: pd.Timedelta | pd.DateOffset, freq: str
+    ) -> pd.PeriodIndex:
+        """Create a PeriodIndex given a start time, timedelta, and frequency."""
+        return pd.period_range(start=start, end=start + timedelta, freq=freq)
+
+    @staticmethod
+    def normalize_row_data(df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
Could switch verbage to `convert_to_returns_data()` and `convert_to_price_data()`

Reply:
This is applied to all columns so returns and price wouldn't make sense - I'm open to changing the name but don't see a strong reason to unless we find a much better one